README.md

```diff
@@ -1,39 +1,120 @@
 # PSI Cardinality
 Private Set Intersection Cardinality protocol based on ECDH and Bloom Filters.
 
+## Requirements
+
+There are requirements for the entire project which each language shares. There also could be requirements for each target language:
+
+### Global Requirements
+These are the common requirements across all target languages of this project.
+
+- [Bazel](https://bazel.build)
+
+
+### Target Requirements
+Some target languages have additional requirements
+
+#### C++
+
+- None
+
+#### JavaScript
+
+- [NodeJS](https://nodejs.org/en/)
+- [Yarn](https://yarnpkg.com/)
+
+
 ## Compiling and Running
-The only requirement is [Bazel](https://bazel.build). To compile, run
+
+The repository uses a folder structure to isolate the supported targets from one another:
+
 ```
-bazel build //...
+src/<target language>/<sources>
```

Comment:
I'm not a big fan of top-level `src` directories, I prefer naming it after the library itself, i.e., `psi_cardinality`.

Also, should we maybe move the `scripts` folder into `javascript`? It's not used anywhere else.

Finally, I think we could move the documentation for each language to their respective subdirectories and just link to them from the top-level README.md. That way, for example someone who's only interested in getting the JS version to run won't have to read through the C++ instructions.